roles/app_docker/tasks/main.yaml

```diff
@@ -0,0 +1,126 @@
+
+- name: Derive docker_app_image variable
+  set_fact:
+    docker_app_image: "{{ docker_app_image_registry }}/{{ docker_app_image_name }}:{{ docker_app_image_version }}"
+  when: docker_app_image is not defined or docker_app_image == None
+
+- name: Create the application group '{{ docker_app_user }}'
+  ansible.builtin.group:
+    name: "{{ docker_app_user }}"
+
+- name: Create the application user
+  ansible.builtin.user:
+    name: "{{ docker_app_user }}"
+    state: present
+    group: "{{ docker_app_user }}"
+    comment: "{{ docker_app_user }} application user"
+    create_home: true
+  register: _docker_app_user
+
+- name: Template out the .env file
+  template:
+    src: "{{ docker_env_template }}"
+    dest: "{{ _docker_app_user.home }}/dotenv"
+    owner: "{{ docker_app_user }}"
+    group: "{{ docker_app_user }}"
+    mode: u=rw,g=r,o=
+  register: _docker_env_file
+
+- name: Add docker volumes
+  community.docker.docker_volume:
+    volume_name: "{{ item.name|mandatory }}"
+  loop: "{{ create_volumes.volumes }}"
+  # This is only needed if we are not mounting an existing directory
+  when: create_volumes.enabled
+
+- name: Provision the isolated app container network
+  community.docker.docker_network:
+    name: "{{ docker_app_name_prefix }}"
+    internal: "{{ docker_network_internal }}"
+
+- name: Copy free ports script to host
+  copy:
+    content: |
+        #!/usr/bin/python3
+        import socket
+        for p in range({{ docker_app_port_range.start }},{{ docker_app_port_range.end }}):
+            import socket
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            try:
+              s.bind(("",p))
+            except:
+                pass
+            port = s.getsockname()[1]
+            s.close()
+            if port != 0:
+              print(p)
+    dest: /tmp/freeports.py
+    mode: 700
```

Comment:
isn't it time to properly package this as an ansible plugin or something like that instead of always copying it over?

Reply:
Good idea! I made an issue for it: https://github.com/maykinmedia/commonground-ansible/issues/47